services/QuillLMS/app/models/question_health_dashboard.rb

```diff
@@ -0,0 +1,74 @@
+class QuestionHealthDashboard
+  MAX_SESSIONS_VIEWED = 500
+
+  def initialize(activity_id, question_number, question_uid)
+    @activity_id = activity_id
+    @question_number = question_number
+    @question_uid = question_uid
+  end
+
+  def average_attempts_for_question
+    failed_attempts = attempt_data.select { |a| a["number_of_attempts"] == '5' && a["correct"] == '0'}
```

Comment:
I really appreciate your clear and methodical variable naming in this file, and the composition of complex queries from simpler queries.